adr/ADR-40.md

```diff
@@ -473,6 +474,32 @@ a command. This is followed by a disconnect.
 - `maximum account active connections exceeded` not notified to the client, the
   client connecting will be disconnected (seen as a connection refused.)
 
+### Client Statistics
+Clients should implement statistics gathered throughout the lifetime of the client (persisted between connections):
+
+#### Bytes In
+Should account for everything received from the server, a record of bytes received on the socket.
```

Comment:
@Jarema - so went looking augment with what is discussed here, but the discussion on the alignment with server is incorrect -  Server accounting seems to only include headers/payload for published messages - it doesn't include other protocol messages, subjects, etc. We could add a `raw` metric for in/out bytes, but right now at least for JavaScript, it lines up with what the server does.

```javascript
Deno.test("basics - stats", async () => {
  const { ns, nc } = await _setup(connect);

  const cid = nc.info?.client_id || -1;
  if (cid === -1) {
    fail("client_id not found");
  }

  await nc.flush();
  console.log(nc.stats())

  console.log(await ns.connz(cid, "detail"));

  nc.publish("hello", "world");
  await nc.flush();

  console.log(nc.stats())
  console.log(await ns.connz(cid, "detail"));


  nc.subscribe("hello", {callback: () => {}});
  nc.publish("hello", "hi");
  await nc.flush();

  console.log(nc.stats())
  console.log(await ns.connz(cid, "detail"));

  await cleanup(ns, nc);
});

```
```javascript
{ inBytes: 0, outBytes: 0, inMsgs: 0, outMsgs: 0 }
{
  server_id: "NBN7DZNKVGIGZBAVMGYTK7RHZZITVELFTCJI5BKOHORHTTO2TOS4OHTL",
  now: "2024-10-15T23:32:44.636996Z",
  num_connections: 1,
  total: 1,
  offset: 0,
  limit: 1,
  connections: [
    {
      cid: 6,
      kind: "Client",
      type: "nats",
      ip: "127.0.0.1",
      port: 59457,
      start: "2024-10-15T18:32:44.633103-05:00",
      last_activity: "2024-10-15T23:32:44.634232Z",
      rtt: "1.129333ms",
      uptime: "0s",
      idle: "0s",
      pending_bytes: 0,
      in_msgs: 0,
      out_msgs: 0,
      in_bytes: 0,
      out_bytes: 0,
      subscriptions: 0,
      lang: "nats.deno",
      version: "3.0.0-6"
    }
  ]
}
{ inBytes: 0, outBytes: 5, inMsgs: 0, outMsgs: 1 }
{
  server_id: "NBN7DZNKVGIGZBAVMGYTK7RHZZITVELFTCJI5BKOHORHTTO2TOS4OHTL",
  now: "2024-10-15T23:32:44.638501Z",
  num_connections: 1,
  total: 1,
  offset: 0,
  limit: 1,
  connections: [
    {
      cid: 6,
      kind: "Client",
      type: "nats",
      ip: "127.0.0.1",
      port: 59457,
      start: "2024-10-15T18:32:44.633103-05:00",
      last_activity: "2024-10-15T18:32:44.638146-05:00",
      rtt: "1.129333ms",
      uptime: "0s",
      idle: "0s",
      pending_bytes: 0,
      in_msgs: 1,
      out_msgs: 0,
      in_bytes: 5,
      out_bytes: 0,
      subscriptions: 0,
      lang: "nats.deno",
      version: "3.0.0-6"
    }
  ]
}
```

Reply:
ok. I probably made a mistake checking the server code.
Though I still agree with your point that we should count everything in and out.
We for sure can have more stats - current one and raw.

Reply:
I think that would be the correct thing to do. The other nuance is we should reset on reconnect the current counters this way the reporting from the client will always match the current session and the server. We can add a `total_in/out` which accounts for all the in/out but these are always all session - Not sure what the benefit is to say that a current client over all its reconnects has sent XX amount of data, so preserving that session data is possibly not useful.